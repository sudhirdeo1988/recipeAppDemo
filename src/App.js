import React, { useState, useEffect } from "react";
import uuid from "react-uuid";
import { Helmet } from "react-helmet";
import { connect } from "react-redux";
import { getRecipeData } from "./utilities/api.js";
import SearchPanel from "./components/SearchPanel/SearchPanel";
import RecipeList from "./components/RecipeList/RecipeList";
import RecipeItemDetails from "./components/RecipeItemDetails/RecipeItemDetails";
import { RECIPE_CONSTANTS } from "./utilities/constants";
import Container from "react-bootstrap/Container";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import debounce from "lodash.debounce";

function App(props) {
  const [isLoading, setLoading] = useState(null);
  const [favState, setfavState] = useState(null);
  const [searchQ, setsearchQ] = useState("");
  const [pageFrom, setpageFrom] = useState(0);
  const [pageTo, setpageTo] = useState(10);

  window.onscroll = debounce(() => {
    if (
      window.innerHeight + document.documentElement.scrollTop ===
      document.documentElement.offsetHeight
    ) {
      setpageFrom(pageFrom + 10);
      setpageTo(pageTo + 10);
    }
  }, 1000);

  const serchQuerydata = (searchQuery) => {
    setsearchQ(searchQuery);
    props.removeRecipe([]);
    fetchRecipeFromAPI(searchQuery, pageFrom, pageTo);
  };

  useEffect(() => {
    if (pageTo - pageFrom === 10) {
      fetchRecipeFromAPI(searchQ, pageFrom, pageTo);
    }
  }, [pageFrom, pageTo]);

  async function fetchRecipeFromAPI(searchQuery, pageFrom, pageTo) {
    setLoading(true);
    const recipeData = await getRecipeData(searchQuery, pageFrom, pageTo);
    const newRecipeData = await recipeData.map(function (el) {
      var o = Object.assign({}, el);
      o.recipe.id = uuid();
      return o.recipe;
    });
    props.addRecipe(newRecipeData);
    setLoading(false);
  }
  const changeState = (st) => {
    setfavState(st);
  };
  return (
    <div className="App">
      <Helmet>
        <title>Recipe App Demo</title>
      </Helmet>
      <SearchPanel
        changeState={changeState}
        serchQuerydata={serchQuerydata}
      ></SearchPanel>

      <Container fluid>
        <Row>
          <Col md={{ span: 8 }}>
            <RecipeList isLoading={isLoading}></RecipeList>
          </Col>
          <Col md={{ span: 4 }}>
            <RecipeItemDetails
              favState={favState}
              changeState
            ></RecipeItemDetails>
          </Col>
        </Row>
      </Container>
    </div>
  );
}

const mapStateToProps = (state) => {
  return {
    recipeInfo: state.recipe,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    addRecipe: (payload) =>
      dispatch({ type: RECIPE_CONSTANTS.ADD.RECIPE, payload }),
    removeRecipe: (payload) =>
      dispatch({ type: RECIPE_CONSTANTS.REMOVE.RECIPE, payload }),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(App);
