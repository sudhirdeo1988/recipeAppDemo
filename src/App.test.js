import React from "react";
import { render, cleanup } from "@testing-library/react";
import configureMockStore from "redux-mock-store";
import thunkMiddleware from "redux-thunk";
import { Provider } from "react-redux";
import App from "./App";

afterEach(cleanup);
test("renders learn react link", () => {
  const initialState = {
    favouriateRecipe: [
      {
        uri:
          "http://www.edamam.com/ontologies/edamam.owl#recipe_09be6cc615db3ab88b48dadc30726fed",
        label: "All-Butter Pie Crust",
        image:
          "https://www.edamam.com/web-img/b50/b5031e0c62073138c6b2f878430337f7.JPG",
        source: "Food52",
        url: "https://food52.com/recipes/6718-all-butter-pie-crust",
        shareAs:
          "http://www.edamam.com/recipe/all-butter-pie-crust-09be6cc615db3ab88b48dadc30726fed/all",
        yield: 2,
        calories: 2651.34,
        totalWeight: 645.0861952394988,
        totalTime: 62,
        id: "5",
      },
    ],
    recipeItem: {
      uri:
        "http://www.edamam.com/ontologies/edamam.owl#recipe_09be6cc615db3ab88b48dadc30726fed",
      label: "All-Butter Pie Crust",
      image:
        "https://www.edamam.com/web-img/b50/b5031e0c62073138c6b2f878430337f7.JPG",
      source: "Food52",
      url: "https://food52.com/recipes/6718-all-butter-pie-crust",
      shareAs:
        "http://www.edamam.com/recipe/all-butter-pie-crust-09be6cc615db3ab88b48dadc30726fed/all",
      yield: 2,
      calories: 2651.34,
      totalWeight: 645.0861952394988,
      totalTime: 62,
      healthLabels: ["Sugar-Conscious", "Vegetarian"],
      id: "4",
    },
    recipe: [
      {
        uri:
          "http://www.edamam.com/ontologies/edamam.owl#recipe_09be6cc615db3ab88b48dadc30726fed",
        label: "All-Butter Pie Crust",
        image:
          "https://www.edamam.com/web-img/b50/b5031e0c62073138c6b2f878430337f7.JPG",
        source: "Food52",
        url: "https://food52.com/recipes/6718-all-butter-pie-crust",
        shareAs:
          "http://www.edamam.com/recipe/all-butter-pie-crust-09be6cc615db3ab88b48dadc30726fed/all",
        yield: 2,
        calories: 2651.34,
        totalWeight: 645.0861952394988,
        totalTime: 62,
        healthLabels: ["Sugar-Conscious", "Vegetarian"],
        id: "4",
      },
    ],
  };
  const mockStore = configureMockStore([thunkMiddleware]);
  const store = mockStore(initialState);
  const { container } = render(
    <Provider store={store}>
      <App />
    </Provider>
  );
  expect(container).toMatchSnapshot();
});
