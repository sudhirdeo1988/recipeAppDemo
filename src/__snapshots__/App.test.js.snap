// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="c-searchPanel"
    >
      <div
        class="c-favouriteList"
      >
        <button
          class="listBtn"
        >
          <i
            class="icon ts-heart-o"
          />
          <span
            class="itemCount"
          >
            1
          </span>
        </button>
      </div>
      <div
        class="container-fluid"
      >
        <div
          class="row"
        >
          <div
            class="col-md-8 offset-md-2"
          >
            <h1
              class="c-pageTitle"
            >
              You will love recipe of our
              <span>
                Delicious menu
              </span>
            </h1>
            <form>
              <div
                class="searchPanel"
              >
                <input
                  class="txtBox"
                  name="companyName"
                  placeholder="Looking For..."
                  type="text"
                />
                <button
                  class="submitBtn"
                  type="submit"
                >
                  Search
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
    <div
      class="container-fluid"
    >
      <div
        class="row"
      >
        <div
          class="col-md-8"
        >
          <div
            class="c-recipeList"
          >
            <h2
              class="sectionHeading"
            >
              Recipe List:
            </h2>
            <div
              class="row"
            >
              <div
                class="col-md-4"
              >
                <div
                  class="c-recipeListItem "
                >
                  <div
                    class="recipeImage"
                  >
                    <div
                      class="recipeType veg"
                    />
                    <img
                      alt="All-Butter Pie Crust"
                      src="https://www.edamam.com/web-img/b50/b5031e0c62073138c6b2f878430337f7.JPG"
                    />
                  </div>
                  <div
                    class="recipeTitle"
                  >
                    All-Butter Pie Crust
                  </div>
                  <div
                    class="recipeBy"
                  >
                    By: 
                    Food52
                  </div>
                  <div
                    class="recipeInfo"
                  >
                    <div
                      class="row"
                    >
                      <div
                        class="col-6"
                      >
                        <label>
                          Calories: 
                        </label>
                        <span>
                          2651.34
                           kcal
                        </span>
                      </div>
                      <div
                        class="col-6"
                      >
                        <label>
                          Total Weight: 
                        </label>
                        <span>
                          645.09
                           g
                        </span>
                      </div>
                    </div>
                  </div>
                  <div
                    class="recipeBtn"
                  >
                    <button
                      class="btnLarge"
                    >
                      Show Details
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="c-loading"
            />
          </div>
        </div>
        <div
          class="col-md-4"
        >
          <div
            class="c-recipeItemDetails"
          >
            <h2
              class="sectionHeading"
            >
              Recipe Details
            </h2>
            <div
              class="r_info"
            >
              <h2
                class="r_name"
              >
                All-Butter Pie Crust
                <button
                  class="btnAdd  "
                  id="btnAdd"
                />
              </h2>
              <div
                class="r_section"
              >
                <label
                  class="s_label"
                >
                  Health Label
                </label>
                <div
                  class="s_block healthLbl"
                >
                  <ul>
                    <li>
                      Sugar-Conscious
                    </li>
                    <li>
                      Vegetarian
                    </li>
                  </ul>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
